### PR TITLE
Station Base Processing Framework

### LINKED ISSUE #3: Station Processing Framework
# **Description**
Framework and base implementation of generic station class for interaction and input of Item class. Design of exact station characteristics and related functionality. Stations include oven, bench, cooktop and cutting board. Stations will spawn in a set location to be aligned with map team. Not handling interaction input.

## SubFeature/component (optional)
- Generic entity for station
- Design of each other station and characteristics

# **Dependencies**
- #4 
- Item Class (Ticket pending upload)

# **Milestones**
- [ ] Set up Entity and related Components for Item intake and output (Aug. 16)
- [ ] Work on documentation for Wiki (Aug. 20)
- [ ] EventHandler and EventListener sections for component calls on specific stations.  Spawn stations in set position (Aug. 23)
- [ ] Test Entity Class

**Completion Deadline:** Aug. 30

# **Documentation**
- [Main description of feature](../wiki/Entity-Component-System-(ECS))
- [User documentation](../wiki/Stations)

# **Members**
- Zach Adams (@AdamsZach)
- Andi Hamilton-Hart (@StrayIntellectual)
- Nicholas Nucifora (@nuciforan0)
- Callan Vragalis (@silagarv)
- Hayden Elliot (@Hayden-Elliott)

### LINKED ISSUE #38: Item Handler Component for Interaction with Inventory Component
# **Description**
**Task:** Item Handler Component for Interaction with Inventory Component
**Feature:** #3 

Create a handler component that calls to inventory component to provide interface for other teams. Improves modularity and reduces need to specify inventory component for each station therefore abstract inventory component. Better plug and play for future changes as well

# **Dependencies**
- [ ] #17 - Will change with updates to inventory component

# **Milestones**
List of steps that need to be completed for this task.
- [x] Set up base framework with calls to Inventory Component (Aug. 21)
- [ ] Ensure capability with calls from PlayerActions Component (Aug. 22)
- [ ] Wiki (Aug. 22)
- [ ] Tests (Aug. 22)
 
**Completion Deadline:** Aug. 23

 
# **Documentation** - Update when complete
- [Main description of feature](../wiki/Game-Screens)
- [JavaDoc](JavaDoc/Location/Package/Class)
- ...

### PR DESCRIPTION
**Pull request by Hayden-Elliott**
Created general structure for individual stations and the general handler component to manage a station inventory. Implementation of StationFactory to facilitate spawning and rendering. Rendered in two different stations. Space and comments left for other groups to work in, review Java Doc comments and inline comments for further details. Wiki in the works.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/station/StationItemHandlerComponent.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    /**
+     *  Called on creation of the station to allow outside interaction within the station.
+     *  Adds the listener for set current item and for remove current item.
+     */
+    @Override
+    public void create() {
+        acceptableItems.add("meat");
+        acceptableItems.add("vegetable");
```
Comment (nimeshgarg) [inline source/core/src/main/com/csse3200/game/components/station/StationItemHandlerComponent.java:42]: Why adding the acceptable items already?

Comment (Hayden-Elliott) [reply source/core/src/main/com/csse3200/game/components/station/StationItemHandlerComponent.java:42]: For testing of data structure, should have been an included comment will add

**Review by silagarv — APPROVED** [review]

**Review by AdamsZach — APPROVED** [review]

**Review by silagarv — APPROVED** [review]
Awesome work team 5!

**Review by nicholasnucifora — APPROVED** [review]
Absolutely spendlid marvelous work my team, keep it up and LETS GOOOOOO

ヽ(￣▽￣)ノミ|Юヽ(￣▽￣)ノミ|Ю(。・ω・。)

**Review by Hayden-Elliott — COMMENTED** [review]
To add comment

**Review by DargonTrolls — COMMENTED** [review]
LGTM!
(｡･∀･)ﾉﾞ

**Review by StrayIntellectual — APPROVED** [review]
LGTM

**Pull request merged**